initializers/logger.js

```diff
@@ -1,14 +1,88 @@
 const bunyan = require('bunyan');
+const stackTrace = require('stacktrace-js');
 
-if (!process.env.LOGGER_NAME) {
-	throw Error('LOGGER_NAME environment variable not provided for logger initializer.');
-}
 
-const logger = bunyan.createLogger({
-	name: process.env.LOGGER_NAME,
-	level: process.env.LOG_LEVEL || 'debug',
-	src: true,
-	stream: process.stdout,
+const DEFAULT_OPTIONS = {
+    name: process.env.LOGGER_NAME,
+    level: process.env.LOG_LEVEL || 'debug',
+    src: true,
+    stream: process.stdout,
+};
+
+const getStack = () => stackTrace.getSync().map(sf => ({
+        func: sf.functionName,
+        file: sf.fileName,
+        line: sf.lineNumber,
+        col: sf.columnNumber
+    }));
+
+const LEVELS = ['fatal', 'error', 'warn', 'info', 'debug', 'trace'];
+
+const createGcLogger = logger => ({
+    child: options => {
+        const child = logger.child(options);
+        const handler = createHandler(child);
+
+        return new Proxy(child, handler);
+    },
+    log: level => function() {
+            if(arguments.length) {
+                let [message, data, meta] = arguments;
+
+                const stack = getStack().slice(2);
+                const { func, file, line, col } = stack[0];
+                const group = data && data.group || `${file}:${line}:${col}`;
+                const subGroup = data && data.subGroup || func;
+
+                return logger[level]({
+                        ...data,
+                        ...meta,
+                        group,
+                        subGroup,
+                        stack,
+                    }, message);
+            } else {
+                return logger[level]();
+            }
+        },
 });
 
-module.exports = logger;
+const createHandler = logger => {
+    const gcLogger = createGcLogger(logger);
+
+    const handler = {
+        get: (target, name) => {
+            if (Object.keys(gcLogger).includes(name)) {
+                return gcLogger[name];
+            }
+
+            if (LEVELS.includes(name)) {
+                return gcLogger.log(name);
+            }
+
+            return target[name];
+        }
+    };
+
+    return handler;
+};
+
+const createLogger = options => {
+    const resolvedOptions = {
+        ...DEFAULT_OPTIONS,
+        ...options
+      };
+
+    if(!resolvedOptions.name) {
+        throw Error('Logger name not provided for logger initializer.');
+    }
+
+    const logger = bunyan.createLogger(resolvedOptions);
+    const handler = createHandler(logger);
+
+    return new Proxy(logger, handler);
```

Comment:
I'm assuming this Proxy is only affecting the instanced of the logger and not the logger class itself.  If so what is the benefit of using the Proxy over creating a custom logging API in which we could seamlessly swap out the underlying logger?